runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/interceptors/FlexibleChecksumsRequestInterceptor.kt

```diff
@@ -36,65 +36,85 @@ import kotlin.coroutines.coroutineContext
  * - If no checksum is configured for the request then use the default checksum algorithm to calculate a checksum.
  *
  * If the request will be streamed:
- * - The checksum calculation is done asynchronously using a hashing & completing body.
+ * - The checksum calculation is done during transmission using a hashing & completing body.
  * - The checksum will be sent in a trailing header, once the request is consumed.
  *
  * If the request will not be streamed:
- * - The checksum calculation is done synchronously
+ * - The checksum calculation is done before transmission
  * - The checksum will be sent in a header
  *
  * Business metrics MUST be emitted for the checksum algorithm used.
  *
  * @param requestChecksumRequired Model sourced flag indicating if checksum calculation is mandatory.
  * @param requestChecksumCalculation Configuration option that determines when checksum calculation should be done.
- * @param userSelectedChecksumAlgorithm The checksum algorithm that the user selected for the request, may be null.
+ * @param requestChecksumAlgorithm The checksum algorithm that the user selected for the request, may be null.
  */
 @InternalApi
-public class FlexibleChecksumsRequestInterceptor(
+public class FlexibleChecksumsRequestInterceptor<I>(
     requestChecksumRequired: Boolean,
     requestChecksumCalculation: HttpChecksumConfigOption?,
-    userSelectedChecksumAlgorithm: String?,
+    requestChecksumAlgorithm: String?,
 ) : AbstractChecksumInterceptor() {
-    private val forcedToCalculateChecksum = requestChecksumRequired || requestChecksumCalculation == HttpChecksumConfigOption.WHEN_SUPPORTED
-    private val checksumHeader = StringBuilder("x-amz-checksum-")
-    private val defaultChecksumAlgorithm = lazy { Crc32() }
-    private val defaultChecksumAlgorithmHeaderPostfix = "crc32"
 
-    private val checksumAlgorithm = userSelectedChecksumAlgorithm?.let {
-        val hashFunction = userSelectedChecksumAlgorithm.toHashFunction()
+    // FIXME: Remove in next minor version bump
```

Comment:
Sure, SDK-KT-385 seems to be more for SDK changes. I can create a ticket for the smith kotlin version bump since it seems like we'll do a minor version bump for the SDK soon and one for smithy kotlin later at a different time

Reply:
I'm pretty sure we will bump both at the same time

Reply:
Okay, in that case I'll just fix the `FIXME`s since this should go into a `v1.4` branch. I have to update the PR.